service/src/main/java/com/ericsson/eiffel/remrem/generate/controller/RemremGenerateController.java

```diff
@@ -78,84 +77,221 @@ public class RemremGenerateController {
+                        JSON_ERROR_STATUS);
+            }
+
+            if (inputData.isJsonArray()) {
+                JsonArray inputEventJsonArray = inputData.getAsJsonArray();
+                for (JsonElement element : inputEventJsonArray) {
+                    JsonObject generatedEvent = (processEvent(msgProtocol, msgType,
+                            failIfMultipleFound, failIfNoneFound, lookupInExternalERs, lookupLimit,
+                            okToLeaveOutInvalidOptionalFields, element.getAsJsonObject()));
+                    generatedEventResults.add(generatedEvent);
+                }
+                boolean success = true;
+                for (JsonElement result : generatedEventResults) {
+                    JsonObject jsonObject = result.getAsJsonObject();
+                    success &= jsonObject.has(META);
                 }
-                if (!parsedResponse.getAsJsonObject().has(RemremGenerateServiceConstants.JSON_ERROR_MESSAGE_FIELD)) {
-                    return new ResponseEntity<>(parsedResponse, HttpStatus.OK);
+                return new ResponseEntity<>(generatedEventResults, success ? HttpStatus.OK : HttpStatus.BAD_REQUEST);
+
+            } else if (inputData.isJsonObject()) {
+                JsonObject inputJsonObject = inputData.getAsJsonObject();
+                JsonObject processedJson = processEvent(msgProtocol, msgType, failIfMultipleFound, failIfNoneFound,
+                        lookupInExternalERs, lookupLimit, okToLeaveOutInvalidOptionalFields, inputJsonObject);
+                HttpStatus status;
+                String statusValue = null;
+                if (processedJson.has(META)) {
+                    status = HttpStatus.OK;
+                    return new ResponseEntity<>(processedJson, status);
+                } else if (processedJson.has(JSON_STATUS_CODE)) {
+                    statusValue = processedJson.get(JSON_STATUS_CODE).toString();
+                    try {
+                        status = HttpStatus.resolve(Integer.parseInt(statusValue));
+                        return new ResponseEntity<>(processedJson, status);
+                    } catch (NumberFormatException e) {
+                        log.error("Invalid status value: '" + statusValue + "' of response " + processedJson);
+                        return createResponseEntity(HttpStatus.BAD_REQUEST, "Invalid status value: '"
+                                + statusValue + "' of response " + processedJson, JSON_ERROR_STATUS);
+                    }
                 } else {
-                    return new ResponseEntity<>(parsedResponse, HttpStatus.BAD_REQUEST);
+                    log.error("There is no status value: '" + statusValue + "' in the response" + processedJson);
+                    return createResponseEntity(HttpStatus.INTERNAL_SERVER_ERROR, "There is no status value: '"
+                            + statusValue + "' in the response " + processedJson, JSON_ERROR_STATUS);
```

Comment:
statusValue is always null here. You should move its declaration from 180 to 185.

Reply:
OK